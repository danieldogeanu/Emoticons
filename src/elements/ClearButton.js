import React from 'react';
import Icon from '../elements/Icon';

function ClearButton() {
	const clearText = 'Clear Input';
	return (
		<button className="ClearButton" title={clearText}>
			<span className="show-for-screen-reader">{clearText}</span>
			<Icon name="close" />
		</button>
	);
}

export default ClearButton;
