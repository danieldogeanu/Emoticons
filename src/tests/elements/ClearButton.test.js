import React from 'react';
import {render} from '@testing-library/react';
import ClearButton from '../../elements/ClearButton';

describe('ClearButton Element', () => {

	it('clear button renders properly', () => {
		const btnTitle = 'Clear Input';
		const {getByTitle, getByText, container} = render(<ClearButton />);
		const renderedBtn = getByTitle(btnTitle);
		const renderedSRTxt = getByText(btnTitle);
		const renderedSVG = container.querySelector('svg');

		// Full Rendered Button
		expect(renderedBtn).toBeInTheDocument();
		expect(renderedBtn).toHaveClass('ClearButton');
		expect(renderedBtn).toContainElement(renderedSRTxt);
		expect(renderedBtn).toContainElement(renderedSVG);
		expect(renderedBtn).toMatchSnapshot();

		// Screen Reader Text
		expect(renderedSRTxt).toBeInTheDocument();
		expect(renderedSRTxt).toHaveClass('show-for-screen-reader');

		// SVG Icon
		expect(renderedSVG).toBeInTheDocument();
		expect(renderedSVG).toHaveClass('icon', 'close');
	});

});
