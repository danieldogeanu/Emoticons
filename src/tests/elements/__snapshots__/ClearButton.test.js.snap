// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton Element clear button renders properly 1`] = `
<button
  class="ClearButton"
  title="Clear Input"
>
  <span
    class="show-for-screen-reader"
  >
    Clear Input
  </span>
  <svg
    class="icon close"
  >
    <use
      xlink:href="#close"
    />
  </svg>
</button>
`;
